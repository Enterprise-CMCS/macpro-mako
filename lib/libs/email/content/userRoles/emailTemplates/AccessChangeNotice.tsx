import { BasicFooter } from "../../email-components";
import { BaseEmailTemplate } from "../../email-templates";
import { statesMap } from "../types";
import { statusMap, userRoleMap } from "../types";

export type UserRoleEmailType = {
  role: keyof typeof userRoleMap;
  status: keyof typeof statusMap;
  applicationEndpointUrl: string;
  territory: keyof typeof statesMap | "N/A";
  fullName: string;
};

export const AccessChangeNoticeEmail = ({ variables }: { variables: UserRoleEmailType }) => {
  const stateAccess = variables.territory === "N/A" ? "" : ` for ${statesMap[variables.territory]}`;
  // TODO: fix - const approvingRole = userRoleMap[approvingUserRole[variables.role]];
  const approvingRole = userRoleMap[variables.role];
  return (
    <BaseEmailTemplate
      previewText={`Your access as a ${variables.role}${stateAccess} has been ${statusMap[variables.status]}.`}
      heading=""
      applicationEndpointUrl={variables.applicationEndpointUrl}
      footerContent={<BasicFooter />}
    >
      <p>Hello,</p>
      <p>
        Your access as a {userRoleMap[variables.role]}
        {stateAccess} has been {statusMap[variables.status]}. If you have any questions, please
        reach out to your {approvingRole}.
      </p>
    </BaseEmailTemplate>
  );
};
