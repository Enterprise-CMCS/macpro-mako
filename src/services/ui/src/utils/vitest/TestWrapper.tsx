import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactLocation, Router } from "@tanstack/react-location";

const queryClient = new QueryClient();
const location = new ReactLocation();

export const TestWrapper = ({ children }: any) => (
  <QueryClientProvider client={queryClient}>
    <Router location={location} routes={[]}>
      {children}
    </Router>
  </QueryClientProvider>
);
