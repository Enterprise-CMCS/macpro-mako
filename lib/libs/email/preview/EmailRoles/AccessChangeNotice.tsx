import { AccessChangeNoticeEmail } from "../../content/userRoles/emailTemplates";
import { userRoleEmailMockVars } from "../../mock-data/email-roles";

export default () => {
  return <AccessChangeNoticeEmail variables={userRoleEmailMockVars} />;
};
