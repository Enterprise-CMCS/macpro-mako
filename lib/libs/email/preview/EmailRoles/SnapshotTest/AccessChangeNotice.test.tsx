import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import AccessChangeNotice from "../AccessChangeNotice";

describe("Access Change Notice Email Snapshot Test", () => {
  it("renders a AppkCMSEmail Preview Template", () => {
    const { asFragment } = render(<AccessChangeNotice />);

    expect(asFragment()).toMatchSnapshot();
  });
});
