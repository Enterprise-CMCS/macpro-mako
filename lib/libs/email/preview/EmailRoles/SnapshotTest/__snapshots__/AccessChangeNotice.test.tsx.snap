// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Access Change Notice Email Snapshot Test > renders a AppkCMSEmail Preview Template 1`] = `
<DocumentFragment>
  <meta
    content="text/html; charset=UTF-8"
    http-equiv="Content-Type"
  />
  <meta
    name="x-apple-disable-message-reformatting"
  />
  <div
    style="display: none; overflow: hidden; line-height: 1px; opacity: 0; max-height: 0; max-width: 0;"
  >
    Your access as a statesystemadmin for Ohio has been pending.
    <div>
       ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿
    </div>
  </div>
  <table
    align="center"
    border="0"
    cellpadding="0"
    cellspacing="0"
    role="presentation"
    style="max-width: 37.5em; background-color: rgb(245, 245, 245);"
    width="100%"
  >
    <tbody>
      <tr
        style="width: 100%;"
      >
        <td>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="background-color: rgb(0, 113, 189); padding: 16px;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <a
                    href="https://mako-dev.cms.gov/"
                    style="color: rgb(6, 125, 247); text-decoration: none; display: block; max-width: 112px;"
                    target="_blank"
                  >
                    <img
                      alt="OneMAC Logo"
                      height="40"
                      src="https://mako-dev.cms.gov/onemac-logo.png"
                      style="max-width: 112px;"
                      width="112"
                    />
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            style="margin: 0px 8px 8px 8px; padding: 8px;"
          >
            <h1
              style="color: rgb(51, 51, 51); font-size: 20px; font-weight: bold; margin-bottom: 15px;"
            />
            <p>
              Hello,
            </p>
            <p>
              Your access as a State System Admin for Ohio has been pending. If you have any questions, please reach out to your CMS Role Approver.
            </p>
            <p
              style="font-size: 14px; line-height: 1.4; margin: 12px 0px; color: rgb(51, 51, 51);"
            >
              Thank you.
            </p>
          </div>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="font-size: 14px; padding: 0px 0px; font-weight: 300; background-color: rgb(0, 113, 189); color: rgb(255, 255, 255); text-align: center;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <table
                    align="center"
                    border="0"
                    cellpadding="0"
                    cellspacing="0"
                    role="presentation"
                    style="font-size: 14px; padding: 0px 0px; font-weight: 300; background-color: rgb(0, 113, 189); color: rgb(255, 255, 255); text-align: center;"
                    width="100%"
                  >
                    <tbody>
                      <tr>
                        <td>
                          <p
                            style="font-size: 14px; line-height: 24px; margin: 8px; color: rgb(255, 255, 255);"
                          >
                            U.S. Centers for Medicare & Medicaid Services
                          </p>
                          <p
                            style="font-size: 14px; line-height: 24px; margin: 8px; color: rgb(255, 255, 255);"
                          >
                            © 2023 | 7500 Security Boulevard, Baltimore, MD 21244
                          </p>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
